nms/packages/magmalte/app/views/equipment/EnodebDetailConfigEdit.js

```diff
@@ -665,6 +928,83 @@ function buildRanConfig(config: enodeb_configuration, optConfig: OptConfig) {
     }
     response['tac'] = parseInt(optConfig.tac);
   }
+  if (optConfig.reference_signal_power !== '') {
+    response.radioConfiguration.powerControlParameters.reference_signal_power = parseInt(
+      optConfig.reference_signal_power,
+    );
+  }
+  if (optConfig.power_class !== '') {
+    response.radioConfiguration.powerControlParameters.power_class = parseInt(
+      optConfig.power_class,
+    );
+  }
+  if (optConfig.pa !== '') {
+    response.radioConfiguration.powerControlParameters.pa = parseInt(
+      optConfig.pa,
+    );
+  }
+  if (optConfig.pb !== '') {
+    response.radioConfiguration.powerControlParameters.pb = parseInt(
+      optConfig.pb,
+    );
+  }
+  if (optConfig.mme_pool_1 !== '') {
+    response['mme_pool_1'] = String(optConfig.mme_pool_1);
+  }
+  if (optConfig.mme_pool_2 !== '') {
+    response['mme_pool_2'] = String(optConfig.mme_pool_2);
+  }
+  if (optConfig.management_server_ssl_enable !== '') {
+    response.managementServer.management_server_ssl_enable = Boolean(
+      optConfig.management_server_ssl_enable,
+    );
+  }
+  if (optConfig.management_server_host !== '') {
+    response.managementServer.management_server_host = String(
+      optConfig.management_server_host,
+    );
+  }
+  if (optConfig.management_server_port !== '') {
+    response.managementServer.management_server_port = parseInt(
+      optConfig.management_server_port,
+    );
+  }
+  if (optConfig.sync_1588_switch !== '') {
+    response.sync_1588.sync_1588_switch = Boolean(optConfig.sync_1588_switch);
+  }
 
+  if (optConfig.sync_1588_asymmetry !== '') {
+    response.sync_1588.sync_1588_asymmetry = parseInt(
+      optConfig.sync_1588_asymmetry,
+    );
+  }
+  if (optConfig.sync_1588_delay_rq_msg_interval !== '') {
+    response.sync_1588.sync_1588_delay_rq_msg_interval = parseInt(
+      optConfig.sync_1588_delay_rq_msg_interval,
+    );
+  }
+  if (optConfig.sync_1588_domain !== '') {
+    response.sync_1588.sync_1588_domain = parseInt(optConfig.sync_1588_domain);
+  }
+  if (optConfig.sync_1588_holdover !== '') {
+    response.sync_1588.sync_1588_holdover = parseInt(
+      optConfig.sync_1588_holdover,
+    );
+  }
+  if (optConfig.sync_1588_msg_interval !== '') {
+    response.sync_1588.sync_1588_msg_interval = parseInt(
+      optConfig.sync_1588_msg_interval,
+    );
+  }
+  if (optConfig.sync_1588_unicast_serverIp !== '') {
+    response.sync_1588.sync_1588_unicast_serverIp = String(
+      optConfig.sync_1588_unicast_serverIp,
+    );
+  }
+  if (optConfig.sync_1588_unicast_enable !== '') {
+    response.sync_1588.sync_1588_unicast_enable = Boolean(
+      optConfig.sync_1588_unicast_enable,
+    );
+  }
```

Comment:
There's a lot of repetitive code added here. Can this be done in a way to adhere to the "DRY" principle?